### ISSUE TITLE
WFS support for non-CRS srsName and multiple dimension geometries

### ISSUE THREAD

**Issue by gbuehler** [gbuehler/7#304]
<b>Posted by:</b>  on 2013-05-28 02:57:20
			    	     <b>Document Title:</b> Web Feature Service 2.0 Interface Standard (also ISO 19142)
                                     <b>Proposal:</b> GML anticipates the use of non-CRS srsName and permits geometries with different srsName in a single instance document, but WFS precludes such patterns, limiting the expressivity of GML. In particular, see the note on non-CRS srsName in section 10.1.3.2 on page 56 of OGC 07-036 (GML 3.2.1).

A specific use-case in which WFS restrictions on encoded geometries is problematic is that of borehole intervals, in which it is desirable that specimens taken from the earth (cores) are georeferenced by their start and end along a curved path (down-hole depth), preserving the relationship between these specimens. GeoSciML is a GML application schema. The GeoSciML community indicate that they would like to represent borehole intervals in GML using a 1D geometry with srsName (an HTTP URI) that references another geometry, the borehole path, rather than a static CRS definition; this appears a natural fit with the information model and encoding available in GML. In effect, every borehole provides its own 1D CRS: the down-hole depth. Another example of such a 1D CRS is a GML 3.3 LinearSRS:
http://www.opengeospatial.org/blog/1654
https://www.seegrid.csiro.au/wiki/CGIModel/BoreHolesAndObservation#Linear_Referencing_40Borehole_41

Section 7.6.5.5 on page 26 and section 7.9.2.4.4 on page 35 and Table 11 on page 48 of OGC 09-025r1 (WFS 2.0.0) require that all geometries in a WFS response be encoded in a single CRS (either DefaultCRS or a member of OtherCRS selected with wfs:Query@srsName). This causes the following problems:

- srsName that references some shape such as a borehole curve cannot be reprojected to reference some other shape. The requirement to encode a response with a particular srsName can only be satisfied if the target srsName is the same as the storage srsName. This is a particular problem for WFS responses that contain multiple boreholes, where each feature contains a geometry with a different srsName; in this case there is no single srsName that can be used for encoding a WFS response.

- WFS responses that contain geometries of different dimensionalities cannot be encoded using a single srsName. For example, a borehole might contain a collar location that is a 2D geometry and also a 3D shape which is the path of the hole itself. The requirement that all WFS responses are encoded in a single CRS precludes the representation of this feature in a WFS response.

In a nutshell: WFS responses are more expressive if some geometries are encoded with srsName other than the target srsName of a request.
